portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -64,13 +69,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    populateList(results);
-
-    Gson gson = new Gson();
+    displayMaxComments(results, this.maxComment);
+    String json = new Gson().toJson(this.comments);
     response.setContentType(APPLICATION_JSON_PARAMETER);
-    response.getWriter().println(gson.toJson(this.comments));
+    response.getWriter().println(json);
+  }
+
+  public void setMaxComment(HttpServletRequest request){
+    String maxComment = request.getParameter(MAXCOMMENT_PARAMETER);
+    this.maxComment = Integer.parseInt(maxComment);
   }
 
+
   public void createEntitys(){
     Entity commentEntity = new Entity(this.COMMENT_PARAMETER);
     commentEntity.setProperty(this.USERNAME_PARAMETER, this.userNameInput);
```

Comment:
`this` isn't needed unless you have a conflicting variable name

Reply:
FWIW I always use this for instance variables.